opentelemetry-sdk/src/metrics/internal/aggregate.rs

```diff
@@ -0,0 +1,185 @@
+use std::{marker, sync::Arc};
+
+use once_cell::sync::Lazy;
+use opentelemetry::KeyValue;
+
+use crate::{
+    metrics::data::{Aggregation, Gauge, Temporality},
+    AttributeSet,
+};
+
+use super::{
+    histogram::Histogram,
+    last_value::LastValue,
+    sum::{PrecomputedSum, Sum},
+    Number,
+};
+
+const STREAM_CARDINALITY_LIMIT: u32 = 2000;
+pub(crate) static STREAM_OVERFLOW_ATTRIBUTE_SET: Lazy<AttributeSet> = Lazy::new(|| {
+    let key_values: [KeyValue; 1] = [KeyValue::new("otel.metric.overflow", "true")];
+    AttributeSet::from(&key_values[..])
+});
+
+/// Checks whether aggregator has hit cardinality limit for metric streams
+pub(crate) fn is_under_cardinality_limit(size: usize) -> bool {
+    size < STREAM_CARDINALITY_LIMIT as usize - 1
```

Comment:
Why we `-1` here? Does it mean 1999 should return `false`?

Reply:
I guess this was added for #1066. Return false for 1999, as the overflow measurements are aggregated in separate metric stream. So total number of metric streams remain 2000.

Reply:
yeah this one is also unchanged from before (aside from removing the unused `&self` param)

Reply:
Got it. We should probably add it to comment to make it clearer